services/121-service/src/registration/dto/registration-status-patch.dto.ts

```diff
@@ -10,9 +11,7 @@ export class RegistrationStatusPatchDto {
   public readonly status: RegistrationStatusEnum;
 
   @ApiProperty({ example: 'Long enough rejection message', required: false })
-  @MinLength(1)
-  @IsString()
-  @IsOptional()
+  @IsNotBothPresent('messageTemplateKey')
```

Comment:
I'm not sure about this, but I think it mostly has to do with the API of the new validator. Would it be possible to build something that explicitly includes both properties?

eg. `  @IsNotBothPresent('message', 'messageTemplateKey')`

I wonder whether this is possible on the DTO itself rather than on a single property, because for me it is weird/unexpected/unintuitive that you only specify a validator on `message`, and you leave `messageTemplateKey` as-is.

Reply:
That would be nicer I tried to find if there is a good option for it but I don't think there is yet: https://github.com/typestack/class-validator/issues/182

Another options would be to just add this validation in the controller: 

![image](https://github.com/global-121/121-platform/assets/34537157/b23a6796-6a35-4a3f-9f2c-74751e431ba2)

Doing it in the DTO is a bit nice because that is where we mostly define the validation of the DTO that comes in. But doing it in the controller is maybe a bit more easy for anyone to understand

Reply:
I think I'd prefer the cleaner solution even if it isn't as comprehensive as the one that you've built. The issue that you've referenced above is an example of what we're lacking here indeed. What would be the downside/risk of doing it in the controller?

Reply:
Validation of the dto object happens in 2 places

Reply:
Had a call about it - decided to go the controller route in this instance.

Reply:
I decided to go the dto route in the end ;)